src/app/shared/image/image-loader.component.ts

```diff
@@ -1,21 +1,42 @@
 import { Component, Input, ElementRef } from '@angular/core';
+import { HalDoc } from '../cms/haldoc';
 
 @Component({
   selector: 'image-loader',
   styleUrls: ['image-loader.component.css'],
   template: `
-    <img *ngIf="src" [src]="src" class="loading" (load)="onLoad()"/>
+    <img *ngIf="src" [src]="src" (load)="onLoad()" (error)="onError()"/>
     `
 })
 
 export class ImageLoaderComponent {
 
   @Input() public src: string;
+  @Input() public imageDoc: HalDoc;
```

Comment:
Yeah, I like the idea of just giving the component the doc it's waiting for.  And assuming our images stay standardized, it will always be under `prx:image -> enclosure`.